cassandra/ZipkinConfigurationLoader.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.cassandra.config.Config;
+import org.apache.cassandra.config.ConfigurationLoader;
+import org.apache.cassandra.config.EncryptionOptions;
+import org.apache.cassandra.config.SeedProviderDef;
+import org.apache.cassandra.exceptions.ConfigurationException;
+
+import java.net.Inet4Address;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public final class ZipkinConfigurationLoader implements ConfigurationLoader {
+
+  public Config loadConfig() throws ConfigurationException {
+    try {
+      Config config = fromConfigYaml();
+      String ip = getHostIpWithoutNSLookup();
+      config.rpc_address = "0.0.0.0";
+      config.listen_address = ip;
+      config.broadcast_rpc_address = ip;
+      Map<String, String> parameters = new LinkedHashMap<>();
+      parameters.put("seeds", ip);
+      config.seed_provider.parameters = parameters;
+      return config;
+    } catch (SocketException e) {
+      throw new ConfigurationException("couldn't get host ip", e);
+    }
+  }
+
+  // http://stackoverflow.com/questions/8765578/get-local-ip-address-without-connecting-to-the-internet
+  private static String getHostIpWithoutNSLookup() throws SocketException {
+    return Collections.list(NetworkInterface.getNetworkInterfaces()).stream()
+        .flatMap(i -> Collections.list(i.getInetAddresses()).stream())
+        .filter(ip -> ip instanceof Inet4Address && ip.isSiteLocalAddress())
+        .findAny().orElseThrow(SocketException::new)
+        .getHostAddress();
+  }
+
+  private static Config fromConfigYaml() {
```

Comment:
This name makes me expect that stuff will be loaded from a yml file, which is exactly what it doesn't do :D How about calling it `defaultConfig` or just adding a comment to explain?